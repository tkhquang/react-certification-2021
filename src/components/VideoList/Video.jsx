import React from 'react';

import {
  StyledVideo,
  StyledVideoCover,
  StyledVideoTitle,
  StyledVideoDescription,
} from './Video.styled';

export default function Video({ title, description, thumbnails }) {
  return (
    <StyledVideo>
      <StyledVideoCover src={thumbnails.high.url} />
      <StyledVideoTitle>{title}</StyledVideoTitle>
      <StyledVideoDescription>{description}</StyledVideoDescription>
    </StyledVideo>
  );
}
