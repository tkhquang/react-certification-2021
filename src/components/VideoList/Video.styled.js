import styled, { css } from 'styled-components';

export const StyledVideo = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.2);
  border-radius: 4px;
`;

export const StyledVideoCover = styled.div`
  height: 200px;
  width: 100%;
  background-color: white;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;

  ${({ src }) => {
    return css`
      background-image: url(${src});
    `;
  }}
`;

export const StyledVideoTitle = styled.div`
  font-size: 24px;
  font-weight: semi-bold;
  margin: 0.75rem;
`;

export const StyledVideoDescription = styled.div`
  margin: 0.75rem;
`;
