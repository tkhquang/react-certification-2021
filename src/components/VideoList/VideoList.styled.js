import styled, { css } from 'styled-components';

import { up } from '../../styles';

export const StyledVideoList = styled.section`
  display: grid;
  grid-template-columns: repeat(1, minmax(0, 1fr));
  gap: 1rem;
  margin: 1rem;

  ${up(
    'xs',
    css`
      grid-template-columns: repeat(2, minmax(0, 1fr));
    `
  )}

  ${up(
    'md',
    css`
      grid-template-columns: repeat(3, minmax(0, 1fr));
    `
  )}

  ${up(
    'lg',
    css`
      grid-template-columns: repeat(4, minmax(0, 1fr));
    `
  )}
`;
