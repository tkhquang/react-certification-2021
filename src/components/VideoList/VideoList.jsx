import React from 'react';

import Video from './Video';
import { StyledVideoList } from './VideoList.styled';

export default function VideoList({ videos, basePath = '' }) {
  if (videos.length === 0) {
    return <span>It is empty</span>;
  }

  return (
    <StyledVideoList>
      {videos.map(({ etag, id, snippet }) => {
        return (
          <Video
            key={etag}
            {...snippet}
            videoId={id?.videoId || id}
            basePath={basePath}
          />
        );
      })}
    </StyledVideoList>
  );
}
