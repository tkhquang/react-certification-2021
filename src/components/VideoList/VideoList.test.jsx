import React from 'react';
import renderer from 'react-test-renderer';

import { dummyData } from '../../test/mocks';

import VideoList from './VideoList';

describe('VideoList', () => {
  test('renders video list correctly', () => {
    const tree = renderer.create(<VideoList videos={dummyData.items} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
