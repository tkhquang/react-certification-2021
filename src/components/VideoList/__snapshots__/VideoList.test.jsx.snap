// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList renders video list correctly 1`] = `
Array [
  <section
    className="sc-eCApnc gzxGuA"
  >
    <a
      className="sc-bdnxRM jKeha-D"
      href="/[object Object]"
      onClick={[Function]}
    >
      <img
        alt="Wizeline"
        className="sc-gtsrHT iYnqJp"
        src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s800-c-k-c0xffffffff-no-rj-mo"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Wizeline
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Wizeline transforms how teams build technology. Its customers accelerate the delivery of innovative products with proven solutions, which combine Wizeline's ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/nmXMgqjQzls"
      onClick={[Function]}
    >
      <img
        alt="Video Tour | Welcome to Wizeline Guadalajara"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/nmXMgqjQzls/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Video Tour | Welcome to Wizeline Guadalajara
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/HYyRZiwBWc8"
      onClick={[Function]}
    >
      <img
        alt="Wizeline Guadalajara | Bringing Silicon Valley to Mexico"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/HYyRZiwBWc8/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Wizeline Guadalajara | Bringing Silicon Valley to Mexico
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Wizeline continues to offer a Silicon Valley culture in burgeoning innovation hubs like Mexico and Vietnam. In 2018, our Guadalajara team moved into a ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/Po3VwR_NNGk"
      onClick={[Function]}
    >
      <img
        alt="Wizeline hace sentir a empleados como en casa"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/Po3VwR_NNGk/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Wizeline hace sentir a empleados como en casa
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        En el 2014, Bismarck fundó Wizeline, compañía tecnológica que trabaja con los corporativos ofreciendo una plataforma para que desarrollen software de forma ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/7PtYNO6g7eI"
      onClick={[Function]}
    >
      <img
        alt="We Are Wizeline"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/7PtYNO6g7eI/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        We Are Wizeline
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Engineering a better tomorrow. Wizeline is a global software development company that helps its clients solve their biggest challenges with design and ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/YuW0CE_8i1I"
      onClick={[Function]}
    >
      <img
        alt="Wizeline Thrives in Mexico City"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/YuW0CE_8i1I/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Wizeline Thrives in Mexico City
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Our vibrant Mexico City office is home to agile software engineers, talented UX designers, and brilliant data scientists. Learn about the rich history of Mexico City.
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/CHzlSGRvWPs"
      onClick={[Function]}
    >
      <img
        alt="Wizeline"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/CHzlSGRvWPs/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Wizeline
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        El plan de Wizeline, una empresa de inteligencia artificial, para ayudar a crecer la comunidad de ciencia de datos en CDMX y todo el país, a través de cursos ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/cjO2fJy8asM"
      onClick={[Function]}
    >
      <img
        alt="A Day in the Life of an Engineering Manager at Wizeline"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/cjO2fJy8asM/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        A Day in the Life of an Engineering Manager at Wizeline
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Fernando Espinoza shares his experience working as an engineering manager at Wizeline and mentoring other engineers. Learn about Fernando's passions ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/zClI9OjgKXM"
      onClick={[Function]}
    >
      <img
        alt="Wizeline Technical Writing Academy | Featuring Eduardo Ocejo"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/zClI9OjgKXM/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Wizeline Technical Writing Academy | Featuring Eduardo Ocejo
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/8bz9R61oY5o"
      onClick={[Function]}
    >
      <img
        alt="Silicon Valley en México"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/8bz9R61oY5o/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Silicon Valley en México
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Empresas de Silicon Valley buscan establecerse en México por el gran talento que hay en nuestro país. Es una investigación de Roberto Domínguez.
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/7dJFraOqcoQ"
      onClick={[Function]}
    >
      <img
        alt="Why Wizeline? featuring Juan Pablo Villa in Mexico City"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/7dJFraOqcoQ/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Why Wizeline? featuring Juan Pablo Villa in Mexico City
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Juan Pablo, known as Gianpa at Wizeline, is a software engineer in our Mexico City office. Gianpa focuses on Android apps, is an integral part of our culture, ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/w-Qwc_XJrWc"
      onClick={[Function]}
    >
      <img
        alt="Wizeline&#39;s 2020 Year in Review"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/w-Qwc_XJrWc/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Wizeline&#39;s 2020 Year in Review
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        There's no doubt that 2020 has been an unprecedented year. However, amidst all the chaos, we achieved remarkable growth in various areas of our business.
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/rjir_cHTl5w"
      onClick={[Function]}
    >
      <img
        alt="Why Wizeline? featuring Hugo Lopez in Mexico City"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/rjir_cHTl5w/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Why Wizeline? featuring Hugo Lopez in Mexico City
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/DcFK1x3NHGY"
      onClick={[Function]}
    >
      <img
        alt="Why Wizeline? (We&#39;re Hiring in Mexico!)"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/DcFK1x3NHGY/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Why Wizeline? (We&#39;re Hiring in Mexico!)
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        A quick look at why people join Wizeline, what motivates us as a team and what it's like to work in our Guadalajara office. Learn more and apply here: ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/3BzYWAqZgFw"
      onClick={[Function]}
    >
      <img
        alt="Why Wizeline? featuring Oswaldo Herrera in Mexico City"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/3BzYWAqZgFw/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Why Wizeline? featuring Oswaldo Herrera in Mexico City
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Oswaldo is a software engineering in Wizeline's Mexico City office. He joined Wizeline because of the camaraderie and deep sense of commitment of our teams.
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/3KVFmT-Tp2w"
      onClick={[Function]}
    >
      <img
        alt="Caso de Éxito AWS: Wizeline [Spanish]"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/3KVFmT-Tp2w/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Caso de Éxito AWS: Wizeline [Spanish]
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Central de socios de APN - https://amzn.to/2S7tIXM Fundada en 2014, Wizeline es una compañía joven e innovadora que nació en la nube para ofrecer soporte ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/aKuPmY2m1Ro"
      onClick={[Function]}
    >
      <img
        alt="Wizeline&#39;s 2019 Year in Review"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/aKuPmY2m1Ro/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Wizeline&#39;s 2019 Year in Review
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/24sTHUyWhRM"
      onClick={[Function]}
    >
      <img
        alt="[1 of 2] Wizeline CEO shares career lessons from Google"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/24sTHUyWhRM/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        [1 of 2] Wizeline CEO shares career lessons from Google
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Founder & CEO Bismarck Lepe on growth opportunities at Wizeline and his career-path experience as an early Google employee. Join our team!
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/IxGc1gSqB3A"
      onClick={[Function]}
    >
      <img
        alt="Wizeline Data Engineering featuring Tania Reyes"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/IxGc1gSqB3A/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Wizeline Data Engineering featuring Tania Reyes
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Tania discovered her interest in Big Data while working at Wizeline and took Wizeline Academy courses to skill up and join the data team. Now, she works on ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/NP1gAnbeNno"
      onClick={[Function]}
    >
      <img
        alt="Wizeline Querétaro | Mexico&#39;s New Knowledge Economy (We&#39;re hiring!)"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/NP1gAnbeNno/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Wizeline Querétaro | Mexico&#39;s New Knowledge Economy (We&#39;re hiring!)
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        A small but mighty (and growing) team, the Queretaro crew has taken ownership of growing the office and brand, speaking at university events, hosting tech ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/F6Krwu6lUc8"
      onClick={[Function]}
    >
      <img
        alt="Miriam Godinez | Women in Leadership at Wizeline"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/F6Krwu6lUc8/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Miriam Godinez | Women in Leadership at Wizeline
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Science and technology always caught Miriam's attention. One of her ultimate goals as a Senior Engineer Manager and Lead from the Mobile Team at Wizeline ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/RFq7gfvhtCk"
      onClick={[Function]}
    >
      <img
        alt="Welcome Back to Wizeline Vietnam | Extended Version"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/RFq7gfvhtCk/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Welcome Back to Wizeline Vietnam | Extended Version
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Thanks to swift government action, the COVID-19 situation in Vietnam has reached a point where businesses are able to return to work and reopen offices.
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/E1Vq_A3WKK8"
      onClick={[Function]}
    >
      <img
        alt="How does Wizeline work?"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/E1Vq_A3WKK8/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        How does Wizeline work?
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Wizeline builds teams with a mix of technical and non-technical talent to deliver better products, faster. Learn more about our consulting services: ...
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/ZmkslANDz0Q"
      onClick={[Function]}
    >
      <img
        alt="12 Wishes from Wizeline | Happy Holidays 2019"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/ZmkslANDz0Q/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        12 Wishes from Wizeline | Happy Holidays 2019
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        
      </div>
    </a>
    <a
      className="sc-bdnxRM jKeha-D"
      href="/Nss3EmTDD3s"
      onClick={[Function]}
    >
      <img
        alt="Why Wizeline?"
        className="sc-gtsrHT iYnqJp"
        src="https://i.ytimg.com/vi/Nss3EmTDD3s/hqdefault.jpg"
      />
      <div
        className="sc-dlnjwi hYfgDE"
      >
        Why Wizeline?
      </div>
      <div
        className="sc-hKFxyN kdhmMD"
      >
        Hear from our employees directly about what excites them about their roles here at Wizeline. Wizeline wants to hire the best and the brightest to accelerate their ...
      </div>
    </a>
  </section>,
  <div
    id="portal"
  />,
]
`;
